### PR TITLE
[USH-3277] allow the geocoder widget to use the user's location

### PR DESCRIPTION
**Pull request by snopoke**
Jira: https://dimagi-dev.atlassian.net/browse/USH-3277

## Product Description
This adds 3 features to the geocoder widget. Each is controlled via a separate toggle (for now):

1. Add a button to the geocder widget that will populate the input with the user's current location (according to the browser).
    
    geolocation-on
    
    If the user does not give permission a message is shown:
    
    geolocation-denied

2. Auto-populate the geocoder widget with the user's location. 

    This is similar to the first point but does not require the user to press the button. When the widget is loaded it will auto-populate with the location of the user.

3. Set the search proximity based on the user's location

    The effect of this is that the search will sort matching locations based on their distance from the user's location.
    This will override the proximity configured for the domain (See "Default project location" setting).

#### Note
The use of feature flags implies that this cannot be configured per app / per app module. If that is desired in the future additional work will be required.

## Technical Summary
The changes are relatively small and use existing features of the geocoder widget. 

## Feature Flag
3 new flags
* `GEOCODER_MY_LOCATION_BUTTON`
  * It is likely that this will be rolled into the `USH_CASE_CLAIM_UPDATES` flag in the future.
* `GEOCODER_AUTOLOAD_USER_LOCATION`
* `GEOCODER_USER_PROXIMITY`

### Safety story
The changes are limited to the geocoder widget in web apps (both case search and form entry).
Local testing was done to ensure the features do not break when enabled / disabled / user denies location access.

### Automated test coverage
None (I may look at this later but leaving it for now)

### QA Plan
None (though I may put this on staging for easier testing by internal team)

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by stephherbers — APPROVED** [review]
Functionality looks good-- the 3 separate FF is a little funky given how they would interact if more than 1 is enabled for example, but based on the request I see why it's done this way for now.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/utils.js [hunk 1]
```diff
@@ -69,23 +70,43 @@ hqDefine("cloudcare/js/form_entry/utils", function () {
...
+        clearCallBack,
+        initialPageData,
+        inputOnKeyDown,
+        showGeolocationButton= false,
+        geolocateOnLoad = false,
+        setProximity = false
+    ) {
+        showGeolocationButton = showGeolocationButton || toggles.toggleEnabled('GEOCODER_MY_LOCATION_BUTTON');
+        geolocateOnLoad = geolocateOnLoad || toggles.toggleEnabled('GEOCODER_AUTOLOAD_USER_LOCATION');
+        setProximity = setProximity || toggles.toggleEnabled('GEOCODER_USER_PROXIMITY');
```
Comment (esoergel) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/utils.js:93]: What's the idea with parameterizing these flags?  That certain instantiations might have them always-on?  Or is that just to make it easier to test in the future?

**Review by esoergel — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/utils.js (hunk 1 above)
Comment (snopoke) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/utils.js:93]: Yea, it mostly for future.

**Comment by snopoke** [comment]
> Functionality looks good-- the 3 separate FF is a little funky given how they would interact if more than 1 is enabled for example, but based on the request I see why it's done this way for now.

They all work fine together, there isn't really any interaction between them.

**Pull request merged**